src/core/data/transform.cc

```diff
@@ -376,6 +404,8 @@ int32_t Delinearize::target_ndim(int32_t source_ndim) const
   return source_ndim - strides_.size() + 1;
 }
 
+void Delinearize::return_promoted_dims(std::vector<int32_t>&) const {};
```

Comment:
```suggestion
void Delinearize::return_promoted_dims(std::vector<int32_t>& dims) const
{
  bool delinearized_dim_was_promoted = false;
  for (auto& promoted : dims) {
    if (promoted == dim_) delinearized_dim_was_promoted = true;
    if (promoted > dim_) promoted += strides_.size() - 1;
  }
  if (delinearized_dim_was_promoted)
    // all the new dims are also therefore considered promoted
    for (int32_t i = 1; i < strides_.size(); ++i) dims.push_back(dim_ + i);
}
```

I think this is what needs to happen here, but @magnatelee should really confirm the logic here (and all the other transforms really).

Reply:
I actually think the partitioning along of the new dimensions created after delinearization is not supported.
I have tried to execute following
```
  a = create_int64_store(shape=(100,))
   a=a.promote(0,10000)
   a=a.delinearize(0,[100,100])
   store_partition = store.partition_by_tiling(100,10,100) #partition by dim 1
   launch_shape = store_partition.partition.color_shape

    task = context.create_manual_task(
        OpCode.COLLECTIVE,
        launch_domain=Rect(launch_shape),
    )
    task.add_input(store_partition)
    task.execute()

```

and got error when from where legate is trying to create partition